fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -388,12 +388,16 @@ struct MetaclusterManagementWorkload : TestWorkload {
 		state bool exists = itr != self->createdTenants.end();
 		state bool hasCapacity = self->createdTenants.size() < self->totalTenantGroupCapacity;
 		state bool retried = false;
+		state bool preferAssignedCluster = deterministicRandom()->coinflip();
 
 		try {
+			state TenantMapEntry createEntry;
+			if (preferAssignedCluster) {
+				createEntry.assignedCluster = self->chooseClusterName();
```

Comment:
If this is set, we should check below that the cluster actually assigned matches the one we chose.

Reply:
This is also making me realize that we need to properly handle the case where someone tries to create the same tenant with a different assigned cluster than where it was previously assigned. We should test for that by occasionally changing the assigned cluster. We can check for and handle this case in `checkForExistingTenant`.

Reply:
is `cluster_not_found` the right error to throw in the check?